qml/utils.js

```diff
@@ -16,3 +16,37 @@ function hashCode(s) {
       h = (h << 5) - h + s.charCodeAt(i++) | 0;
   return h;
 }
+
+/**
+ * Add `value` to `arr` if `arr` does not already include `value`.
+ *
+ * @param arr an array
+ * @param value a value to add to `arr` if not already present
+ *
+ * @return a new array that is the union of `arr` and `[value]` - unless `arr`
+ * already includes `value`, in which case `arr` is returned unchanged.
+ */
+function including(arr, value) {
+    if (includes(arr, value)) {
+        return arr;
+    } else {
+        return arr.concat(value);
+    }
+}
+
+function includes(arr, value) {
```

Comment:
I think these are not needed, we should be able to use JS `includes` function directly, since we are on a newver version of Qt in this project